core/src/contracts/pool.rs

```diff
@@ -4,10 +4,21 @@ use ergo_lib::ergotree_ir::chain::token::TokenId;
 use ergo_lib::ergotree_ir::ergo_tree::ErgoTree;
 use ergo_lib::ergotree_ir::mir::constant::TryExtractInto;
 
+use thiserror::Error;
+
+#[derive(Clone)]
 pub struct PoolContract {
     ergo_tree: ErgoTree,
 }
 
+#[derive(Debug, Error)]
+pub enum PoolContractError {
+    #[error("pool contract: failed to get update NFT from constants")]
+    NoUpdateNftId,
+    #[error("pool contract: failed to get refresh NFT from constants")]
+    NoRefreshNftId,
+}
+
 impl PoolContract {
     // via
     // https://wallet.plutomonkey.com/p2s/?source=ewogIC8vIFRoaXMgYm94IChwb29sIGJveCkKICAvLyAgIGVwb2NoIHN0YXJ0IGhlaWdodCBpcyBzdG9yZWQgaW4gY3JlYXRpb24gSGVpZ2h0IChSMykKICAvLyAgIFI0IEN1cnJlbnQgZGF0YSBwb2ludCAoTG9uZykKICAvLyAgIFI1IEN1cnJlbnQgZXBvY2ggY291bnRlciAoSW50KQogIC8vICAgdG9rZW5zKDApIHBvb2wgdG9rZW4gKE5GVCkKICAKICB2YWwgb3RoZXJUb2tlbklkID0gSU5QVVRTKDEpLnRva2VucygwKS5fMQogIHZhbCByZWZyZXNoTkZUID0gZnJvbUJhc2U2NCgiVkdwWGJscHlOSFUzZUNGQkpVUXFSeTFMWVU1a1VtZFZhMWh3TW5NMWRqZz0iKSAvLyBUT0RPIHJlcGxhY2Ugd2l0aCBhY3R1YWwKICB2YWwgdXBkYXRlTkZUID0gZnJvbUJhc2U2NCgiWWxGbFZHaFhiVnB4TkhRM2R5RjZKVU1xUmkxS1FFNWpVbVpWYWxodU1uST0iKSAvLyBUT0RPIHJlcGxhY2Ugd2l0aCBhY3R1YWwKCiAgc2lnbWFQcm9wKG90aGVyVG9rZW5JZCA9PSByZWZyZXNoTkZUIHx8IG90aGVyVG9rZW5JZCA9PSB1cGRhdGVORlQpCn0=
```

Comment:
Scalahub added some comments to the contract in [v2.0a](https://github.com/scalahub/OraclePool/blob/58726e7751b2ff454ce4b3f45315f3cd549c2994/src/main/scala/oraclepool/v2a/Contracts.scala#L31-L34). I've made the changes on the ergo playground:
```suggestion
    // https://wallet.plutomonkey.com/p2s/?source=ewogIC8vIFRoaXMgYm94IChwb29sIGJveCkKICAvLyAgIGVwb2NoIHN0YXJ0IGhlaWdodCBpcyBzdG9yZWQgaW4gY3JlYXRpb24gSGVpZ2h0IChSMykKICAvLyAgIFI0IEN1cnJlbnQgZGF0YSBwb2ludCAoTG9uZykKICAvLyAgIFI1IEN1cnJlbnQgZXBvY2ggY291bnRlciAoSW50KQogIC8vCiAgLy8gICB0b2tlbnMoMCkgcG9vbCB0b2tlbiAoTkZUKQogIC8vICAgdG9rZW5zKDEpIHJld2FyZCB0b2tlbnMKICAvLyAgIFdoZW4gaW5pdGlhbGl6aW5nIHRoZSBib3gsIHRoZXJlIG11c3QgYmUgb25lIHJld2FyZCB0b2tlbi4gV2hlbiBjbGFpbWluZyByZXdhcmQsIG9uZSB0b2tlbiBtdXN0IGJlIGxlZnQgdW5jbGFpbWVkCiAgCiAgdmFsIG90aGVyVG9rZW5JZCA9IElOUFVUUygxKS50b2tlbnMoMCkuXzEKICB2YWwgcmVmcmVzaE5GVCA9IGZyb21CYXNlNjQoIlZHcFhibHB5TkhVM2VDRkJKVVFxUnkxTFlVNWtVbWRWYTFod01uTTFkamc9IikgLy8gVE9ETyByZXBsYWNlIHdpdGggYWN0dWFsCiAgdmFsIHVwZGF0ZU5GVCA9IGZyb21CYXNlNjQoIllsRmxWR2hYYlZweE5IUTNkeUY2SlVNcVJpMUtRRTVqVW1aVmFsaHVNbkk9IikgLy8gVE9ETyByZXBsYWNlIHdpdGggYWN0dWFsCgogIHNpZ21hUHJvcChvdGhlclRva2VuSWQgPT0gcmVmcmVzaE5GVCB8fCBvdGhlclRva2VuSWQgPT0gdXBkYXRlTkZUKQp9
```

Reply:
Good catch! Thank you!